proxy/src/main/java/com/wavefront/agent/listeners/DataDogPortUnificationHandler.java

```diff
@@ -306,6 +306,13 @@ private boolean reportMetric(final JsonNode metric, @Nullable final AtomicIntege
       if (deviceNode != null) {
         tags.put("device", deviceNode.textValue());
       }
+      int interval = 1; // If the metric is of type rate its value needs to be multiplied by the specified interval
+      JsonNode type = metric.get("type");
+      if (type != null) {
+        if (type.textValue().equals("rate")) {
+          interval = metric.get("interval") == null ? 1 : metric.get("interval").intValue();
```

Comment:
We probably want to handle bad data scenarios where "interval" field is not numeric

Reply:
ok, done